src/crimson/osd/pg_backend.h

```diff
@@ -129,18 +126,12 @@ class PGBackend
   struct loaded_object_md_t {
     ObjectState os;
     std::optional<SnapSet> ss;
+    using ref = std::unique_ptr<loaded_object_md_t>;
   };
-  load_metadata_ertr::future<loaded_object_md_t> load_metadata(
+  load_metadata_ertr::future<loaded_object_md_t::ref> load_metadata(
     const hobject_t &oid);
 
 private:
-  using cached_ss_t = boost::local_shared_ptr<SnapSet>;
-  SharedLRU<hobject_t, SnapSet> ss_cache;
-  get_os_errorator::future<cached_ss_t> _load_ss(const hobject_t& oid);
-
-  SharedLRU<hobject_t, ObjectState> os_cache;
```

Comment:
if we believe that `intrusive_lru` is better than `SharedLRU`, probably we could just drop the latter? also why it's better? could you note this down in the commit message?

Reply:
Refactoring a user to use intrusive_lru instead would take work, no reason to go and do that to all existing users.

intrusive_lru is only better in the sense that it avoids extra allocations as the map and list headers are embedded in the object itself.

Reply:
since we have a `LRUCache` (in `src/crimson/common/shared_lru.h`), which is only designed for crimson. i suggest remove it in this PR as a separated commit. as it's not needed anymore.

Reply:
Will do.

Reply:
Hadn't realized that crimson/common had its own copy.